File: src/main/java/c6h2cl2/mod/SolidXp/Util/RecipeUtil.java
package c6h2cl2.mod.SolidXp.Util;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.CraftingManager;
import net.minecraftforge.fml.common.registry.GameRegistry;

/**
 * Created by ma87k on 2016/07/10.
 */
public class RecipeUtil {

    public static void registerSwordRecipe(Item output, ItemStack ingot, ItemStack rod){
        GameRegistry.addRecipe(new ItemStack(output)," I "," I "," R ",'I',ingot,'R',rod);
    }
    public static void registerSwordRecipe(Item output, Item ingot, ItemStack rod){
        registerSwordRecipe(output,new ItemStack(ingot),rod);
    }
    public static void registerSwordRecipe(Item output, ItemStack ingot, Item rod){
        registerSwordRecipe(output,ingot,new ItemStack(rod));
    }
    public static void registerSwordRecipe(Item output, Item ingot, Item rod){
        registerSwordRecipe(output,new ItemStack(ingot),new ItemStack(rod));
    }
    public static void registerPickaxeRecipe(Item output, ItemStack ingot, ItemStack rod){
        GameRegistry.addRecipe(new ItemStack(output),"III"," R "," R ",'I',ingot,'R',rod);
    }
    public static void registerPickaxeRecipe(Item output, Item ingot, Item rod){
        registerPickaxeRecipe(output,new ItemStack(ingot),new ItemStack(rod));
    }
    public static void registerAxeRecipe(Item output, Item ingot, Item rod){
        GameRegistry.addRecipe(new ItemStack(output),"II ","IR "," R ",'I',ingot,'R',rod);
    }
    public static void registerShovelRecipe(Item output, Item ingot, Item rod){
        GameRegistry.addRecipe(new ItemStack(output)," I "," R "," R ",'I',ingot,'R',rod);
    }
    public static void registerHoeRecipe(Item output, Item ingot, Item rod){
        GameRegistry.addRecipe(new ItemStack(output),"II "," R "," R ",'I',ingot,'R',rod);
    }

}

File: src/main/kotlin/c6h2cl2/mod/SolidXp/Util/RecipeUtil.kt
package c6h2cl2.mod.SolidXp.Util

import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.CraftingManager
import net.minecraftforge.fml.common.registry.GameRegistry

/**
 * Created by ma87k on 2016/07/10.
 */
object RecipeUtil {

    fun registerSwordRecipe(output: Item, ingot: ItemStack, rod: ItemStack) {
        GameRegistry.addRecipe(ItemStack(output), " I ", " I ", " R ", 'I', ingot, 'R', rod)
    }

    fun registerSwordRecipe(output: Item, ingot: Item, rod: ItemStack) {
        registerSwordRecipe(output, ItemStack(ingot), rod)
    }

    fun registerSwordRecipe(output: Item, ingot: ItemStack, rod: Item) {
        registerSwordRecipe(output, ingot, ItemStack(rod))
    }

    fun registerSwordRecipe(output: Item, ingot: Item, rod: Item) {
        registerSwordRecipe(output, ItemStack(ingot), ItemStack(rod))
    }

    fun registerPickaxeRecipe(output: Item, ingot: ItemStack, rod: ItemStack) {
        GameRegistry.addRecipe(ItemStack(output), "III", " R ", " R ", 'I', ingot, 'R', rod)
    }

    fun registerPickaxeRecipe(output: Item, ingot: Item, rod: Item) {
        registerPickaxeRecipe(output, ItemStack(ingot), ItemStack(rod))
    }

    fun registerAxeRecipe(output: Item, ingot: Item, rod: Item) {
        GameRegistry.addRecipe(ItemStack(output), "II ", "IR ", " R ", 'I', ingot, 'R', rod)
    }

    fun registerShovelRecipe(output: Item, ingot: Item, rod: Item) {
        GameRegistry.addRecipe(ItemStack(output), " I ", " R ", " R ", 'I', ingot, 'R', rod)
    }

    fun registerHoeRecipe(output: Item, ingot: Item, rod: Item) {
        GameRegistry.addRecipe(ItemStack(output), "II ", " R ", " R ", 'I', ingot, 'R', rod)
    }

}
